lib/modules/manager/azure-pipelines/extract.ts

```diff
@@ -178,7 +193,7 @@ export function extractPackageFile(
   }
 
   for (const repository of coerceArray(pkg.resources?.repositories)) {
-    const dep = extractRepository(repository);
+    const dep = extractRepository(repository, config.repository);
```

Comment:
ha. didn't know that current repo was already passed 🙃

Reply:
So the `repository` parameter is the repository that contains the Azure Pipelines templates referenced from the `config.repository`. The new `config.repository`proveds access to the details of the repository being processed by renovate which references the pipeline templates in the repository provided by the `repository` parameter.